### PR TITLE
Ft/colors selection

### PR DESCRIPTION
**Pull request by nikunjsdev**
https://github.com/adrienshen/saleor-storefront/projects/1#card-39394335
colors

### CODE REVIEW

**Comment by zshanhui** [comment]
@nikunjsdev 

After reviewing the implementation. I think it might be better to store `cabinetColors` in `localStorage` this way:

```
{
  [CollectionID!]: {
    top: ID!;
    bottom: ID!;
  },
  ...otherCollections
}
```
This way, it will accommodate for multiple collections.

What do you think?

File: src/@next/components/molecules/ColorSelection/ColorSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import { IProps, IColor, LocalStorageKeys, IPosition } from "./types";
+import { ColorItem } from "./ColorItem";
+import "./index.scss";
+
+const getDataFromStorage = () =>
```
Comment (zshanhui) [inline src/@next/components/molecules/ColorSelection/ColorSelection.tsx]: There should already be a react hook you can use for this. `useLocalStorage`, see if that would work for your use case first.

File: src/@next/components/molecules/ColorSelection/ColorSelection.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  const onChangeColor = (type: IPosition, colorID: number) => {
+    const color = {
+      ...selectedColor,
+      [type]: colorID,
+      collectionId: collectionId,
+    };
+    localStorage.setItem(LocalStorageKeys.CabinetColor, JSON.stringify(color));
+    setSelectedColor(color);
+  };
```
Comment (zshanhui) [inline src/@next/components/molecules/ColorSelection/ColorSelection.tsx:26]: I left a comment on the main page about how we can store this to be more flexible.

File: src/views/CollectionIndex/Colors/Page.tsx [hunk 3]
```diff
@@ -1,5 +1,85 @@
...
+    image: { url: color2, alt: "" },
+    position: IPosition.Bottom,
+  },
+  {
+    id: 11,
+    name: "Dark Elegance",
+    image: { url: color3, alt: "" },
+    position: IPosition.Bottom,
+  },
+];
```
Comment (zshanhui) [inline src/views/CollectionIndex/Colors/Page.tsx]: Can we move these to another file like `mocks.js`?

**Review by zshanhui — CHANGES_REQUESTED** [review]

**Comment by zshanhui** [comment]
@nikunjsdev 

TSLint error in CircleCI:

```
#!/bin/bash -eo pipefail
npm run tslint

> saleor-site@0.7.1 tslint /home/circleci/app
> tslint 'src/**/*.ts?(x)'


Could not find implementations for the following rules specified in the configuration:
    prettier
Try upgrading TSLint and/or ensuring that you have all necessary custom rules installed.
If TSLint was recently upgraded, you may have old rules configured which need to be cleaned up.
        
Warning: The 'no-chainable-assignment' rule requires type information.
Warning: The 'no-chainable-arguments' rule requires type information.

ERROR: src/@next/components/molecules/ColorSelection/ColorSelection.tsx:20:7 - Expected property shorthand in object literal ('{collectionId}').

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! saleor-site@0.7.1 tslint: `tslint 'src/**/*.ts?(x)'`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the saleor-site@0.7.1 tslint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/circleci/.npm/_logs/2020-07-03T19_32_46_597Z-debug.log

Exited with code exit status 2
```

**Comment by nikunjsdev** [comment]
> @nikunjsdev
> 
> After reviewing the implementation. I think it might be better to store `cabinetColors` in `localStorage` this way:
> 
> ```
> {
>   [CollectionID!]: {
>     top: ID!;
>     bottom: ID!;
>   },
>   ...otherCollections
> }
> ```
> 
> This way, it will accommodate for multiple collections.
> 
> What do you think?

@adrienshen 
right . I have done that.

File: src/views/CollectionIndex/Colors/Page.tsx (hunk 3 above)
Comment (nikunjsdev) [reply src/views/CollectionIndex/Colors/Page.tsx]: Done

File: src/@next/components/molecules/ColorSelection/ColorSelection.tsx (hunk 2 above)
Comment (nikunjsdev) [reply src/@next/components/molecules/ColorSelection/ColorSelection.tsx:26]: Done

File: src/@next/components/molecules/ColorSelection/ColorSelection.tsx (hunk 1 above)
Comment (nikunjsdev) [reply src/@next/components/molecules/ColorSelection/ColorSelection.tsx]: Done

**Review by zshanhui — APPROVED** [review]

**Comment by zshanhui** [comment]
@nikunjsdev It's approved, but can you check why CircleCI build not pass?

**Comment by nikunjsdev** [comment]
> @nikunjsdev It's approved, but can you check why CircleCI build not pass?

@adrienshen Above CircleCI issue is resolved in last commit. Is there any new CircleCI issue getting after last commit ?

**Pull request merged**